Client/Frontend/Settings/SettingsTableViewController.swift

```diff
@@ -194,6 +194,135 @@ class BoolSetting: Setting {
+
+    let prefKey: String
+
+    private let prefs: Prefs
+    private let defaultValue: String?
+    private let placeholder: String?
+    private let settingDidChange: ((String?) -> Void)?
+    private let settingIsValid: ((String?) -> Bool)?
+
+    let textField = UITextField()
+
+    init(prefs: Prefs, prefKey: String, defaultValue: String? = nil, placeholder: String? = nil, accessibilityIdentifier: String? = nil, settingIsValid isValueValid: ((String?) -> Bool)? = nil, settingDidChange: ((String?) -> Void)? = nil) {
+        self.prefs = prefs
+        self.prefKey = prefKey
+        self.defaultValue = defaultValue
+        self.settingDidChange = settingDidChange
+        self.settingIsValid = isValueValid
+        self.placeholder = placeholder
+
+        super.init()
+        self.accessibilityIdentifier = accessibilityIdentifier
+    }
+
+    override func onConfigureCell(cell: UITableViewCell) {
+        super.onConfigureCell(cell)
+        if let id = accessibilityIdentifier {
+            textField.accessibilityIdentifier = id + "TextField"
+        }
+        textField.placeholder = placeholder
+        textField.delegate = self
+        textField.addTarget(self, action: #selector(textFieldDidChange), forControlEvents: .EditingChanged)
+
+        cell.accessibilityTraits = UIAccessibilityTraitLink
+        cell.contentView.addSubview(textField)
+
+        let container = UIView()
+        container.addSubview(textField)
+        cell.contentView.addSubview(container)
+
+        cell.contentView.snp_makeConstraints { make in
+            make.height.equalTo(44)
+            make.width.equalTo(cell.snp_width)
+        }
+
+        textField.snp_makeConstraints { make in
+            make.height.equalTo(cell.contentView)
+            make.width.equalTo(cell.contentView).offset(-2 * SettingsTableSectionHeaderFooterViewUX.titleHorizontalPadding)
+            make.leading.equalTo(cell.contentView).offset(SettingsTableSectionHeaderFooterViewUX.titleHorizontalPadding)
+        }
+        textField.text = prefs.stringForKey(prefKey) ?? defaultValue
+        textFieldDidChange(textField)
+    }
+
+    override func onClick(navigationController: UINavigationController?) {
+        textField.becomeFirstResponder()
+    }
+
+    private func isValid(value: String?) -> Bool {
+        return settingIsValid?(value) ?? true
+    }
+
+    @objc func textFieldDidChange(textField: UITextField) {
+        let color = isValid(textField.text) ? UIConstants.TableViewRowTextColor : UIConstants.DestructiveRed
+        textField.textColor = color
+    }
+
+    @objc func textFieldShouldReturn(textField: UITextField) -> Bool {
+        return isValid(textField.text)
+    }
+
+    @objc func textFieldDidEndEditing(textField: UITextField) {
+        let text = textField.text
+        if !isValid(text) {
+            return
+        }
+        // unsure if we want to actually set the value to the default here.
```

Comment:
To answer your comment, I'd prefer just removing the key like you're doing here. That way, if we decide to change the default in the future, we don't have to worry about stale defaults being saved.